pymc_marketing/clv/distributions.py

```diff
@@ -293,3 +294,193 @@ def logp(value, lam, p, T, T0):
+        {(\alpha +T)^{r+s+x}}] \\
+        \\
+        \text{if }\beta >= \alpha: \\
+        \\
+        \mathbb{L}(r, \alpha, s, \beta | x, t_x, T) &=
+        \frac{\Gamma(r+x)\alpha^r\beta}{\Gamma(r)+(\beta +t_x)^{r+s+x}}
+        [(\frac{s}{r+s+x})_2F_1(r+s+x,r+x;r+s+x+1;\frac{\beta-\alpha}{\beta+t_x}) \\
+        &+ (\frac{r+x}{r+s+x})
+        \frac{_2F_1(r+s+x,r+x+1;r+s+x+1;\frac{\beta-\alpha}{\beta+T})(\beta +t_x)^{r+s+x}}
+        {(\beta +T)^{r+s+x}}]
+        \end{align}
+
+    ========  ===============================================
+    Support   :math:`t_j > 0` for :math:`j = 1, \dots, x`
+    Mean      :math:`\mathbb{E}[X(t) | r, \alpha, s, \beta] = \frac{r\beta}{\alpha(s-1)}[1-(\frac{\beta}{\beta + t})^{s-1}]`
+    ========  ===============================================
+
+    References
+    ----------
+    .. [2] David C. Schmittlein, Donald G. Morrison and Richard Colombo.
+           "Counting Your Customers: Who Are They and What Will They Do Next."
+           Management Science,Vol. 33, No. 1 (Jan., 1987), pp. 1-24.
+    
+    .. [3] Fader, Peter & G. S. Hardie, Bruce (2005).
+           "A Note on Deriving the Pareto/NBD Model and Related Expressions."
+           http://brucehardie.com/notes/009/pareto_nbd_derivations_2005-11-05.pdf
+    """
+    rv_op = pareto_nbd
+
+    @classmethod
+    def dist(cls, r, alpha, s, beta, T, **kwargs):
+        return super().dist([r, alpha, s, beta, T], **kwargs)
+
+    def logp(value, r, alpha, s, beta, T):
+        t_x = value[..., 0]
+        x = value[..., 1]
+
+        rsx = r + s + x
+        rx = r + x
+
+        switch_cond = pt.gt(alpha, beta)
+        gt_param = pt.switch(switch_cond, alpha, beta)
+        param_diff = pt.switch(switch_cond, alpha - beta, beta - alpha)
+        hyp2f1_param1 = pt.switch(switch_cond, s + 1, rx)
+        hyp2f1_param2 = pt.switch(switch_cond, s, rx + 1)
+
+        # This term is factored out of the denominator of hyp2f_t1 for numerical stability
+        refactored_term = rsx * pt.log(gt_param + t_x)
+
+        hyp2f1_t1 = pt.log(
+            pt.hyp2f1(rsx, hyp2f1_param1, rsx + 1, (param_diff) / (gt_param + t_x))
+        )
+        hyp2f1_t2 = (
+            pt.log(
+                pt.hyp2f1(rsx, hyp2f1_param2, rsx + 1, (param_diff) / (gt_param + T))
+            )
+            - rsx * pt.log(gt_param + T)
+            + refactored_term
+        )
+
+        A1 = (
+            pt.gammaln(rx)
+            - pt.gammaln(r)
+            + r * pt.log(gt_param)
+            + s * pt.log(beta)
+            - refactored_term
+        )
+        A2 = pt.log(s) - pt.log(rsx) + hyp2f1_t1
+        A3 = pt.log(rx) - pt.log(rsx) + hyp2f1_t2
+
+        logp = A1 + pt.logaddexp(A2, A3)
+
+        logp = pt.switch(
+            pt.any(
+                (
+                    pt.lt(t_x, 0),
+                    pt.lt(x, 0),
+                    pt.gt(t_x, T),
+                ),
+            ),
+            -np.inf,
+            logp,
+        )
```

Comment:
Shouldn't use `any` to not collapse invalid observations with valid observations

```suggestion
        logp = pt.switch(
            pt.or_(
                pt.or_(
                    pt.lt(t_x, 0),
                    pt.lt(x, 0),
                ),
                pt.gt(t_x, T),
            ),
            -np.inf,
            logp,
        )
```

Reply:
Change made in commit 59eda77